samples/sample_multi_transcode/src/pipeline_transcode.cpp

```diff
@@ -3046,7 +3046,8 @@ mfxStatus CTranscodingPipeline::InitVppMfxParams(sInputParams *pInParams)
             break;
         default:
             assert(0);
-            MSDK_CHECK_STATUS(MFX_ERR_UNSUPPORTED, "Unexpected encoder FourCC");
+			sts = MFX_ERR_UNSUPPORTED;
+            MSDK_CHECK_STATUS(sts, "Unexpected encoder FourCC");
```

Comment:
Sometimes we want to use such constructions, so you just introduce unnecessary workaround to fool the compiler. I don't like this particular change

Reply:
yes, this one looks ugly. Which warning you have tried to fix? I don't quite see why compiler cries out....

BTW, please, fix your editor settings: we try to avoid using tabs and I think they squeezed thru in line 3849.

Reply:
I have tried to fix C4127 warning: conditional expression is constant

Reply:
I see. @mgonchar : do you have an idea how to fix it by some macro change? According to https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4127?view=vs-2019 compiler will skip the warning for some trivial expressions like `if (1)` or if for const expression. Maybe something like that:

```
#define MSDK_CHECK_STATUS(X, MSG) \
{ \
    const bool _if_expr = (X) < MFX_ERR_NONE;
    if (_if_expr) \
    {MSDK_PRINT_RET_MSG(X, MSG); return X;} \
}
```

If I read MSFT description correctly, this should work starting from VS2015 update 3. Which VS we use in github mediasdk?

Reply:
I would say it's better to use ```constexpr```:
```
#define MSDK_CHECK_STATUS(X, MSG) \
{ \
    constexpr bool sts_is_error = (X) < MFX_ERR_NONE; \
    if (sts_is_error) \
    {MSDK_PRINT_RET_MSG(X, MSG); return X;} \
}
```

According to .sln files in repo it is VS15

Reply:
With `constexpr` compiler gives` C2131` error: `expression did not evaluate to a constant`, so I used `const bool`